From 642a6e9d81215e7060fefae63f169ffdb5925e12 Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Fri, 25 May 2018 11:25:16 +0800
Subject: [PATCH 1932/2328] media: intel-ipu4: fix logical bitwise operator
 issue

Message for Open Source:
This issue was reported by clang-6.0 compile:
ipu-isys-csi2-be-soc.c:96:8: warning: logical not is only applied to
the left hand side of this bitwise operator
[-Wlogical-not-parentheses]

Message for Internal:
[Issue/Feature] Same as above.
[Root Cause/Changes] Fix this obvious bug.

Change-Id: I4d3052330194f3925d85656385d0e88504bfe7f1
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-isys-csi2-be-soc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu-isys-csi2-be-soc.c b/drivers/media/pci/intel/ipu-isys-csi2-be-soc.c
index fd3c5a8..b0f9739 100644
--- a/drivers/media/pci/intel/ipu-isys-csi2-be-soc.c
+++ b/drivers/media/pci/intel/ipu-isys-csi2-be-soc.c
@@ -92,7 +92,8 @@ ipu_isys_csi2_be_soc_set_sel(struct v4l2_subdev *sd,
 		int i;
 
 		for (i = 0; i < asd->nstreams; i++) {
-			if (!asd->route[i].flags & V4L2_SUBDEV_ROUTE_FL_ACTIVE)
+			if (!(asd->route[i].flags &
+			      V4L2_SUBDEV_ROUTE_FL_ACTIVE))
 				continue;
 			if (asd->route[i].source == sel->pad) {
 				sink_pad = asd->route[i].sink;
-- 
2.7.4

